packages/aws-cdk-lib/aws-eks/lib/cluster.ts

```diff
@@ -1677,12 +1737,23 @@ export class Cluster extends ClusterBase {
       new CfnOutput(this, 'ClusterName', { value: this.clusterName });
     }
 
+    const supportAuthenticationApi = (this.authenticationMode === AuthenticationMode.API ||
+      this.authenticationMode === AuthenticationMode.API_AND_CONFIG_MAP) ? true : false;
+
     // do not create a masters role if one is not provided. Trusting the accountRootPrincipal() is too permissive.
     if (props.mastersRole) {
       const mastersRole = props.mastersRole;
 
-      // map the IAM role to the `system:masters` group.
-      this.awsAuth.addMastersRole(mastersRole);
+      // if we support authentication API we create access entry
+      if (supportAuthenticationApi) {
+        this.grantAccess('mastersRoleAccess', props.mastersRole.roleArn, [
```

Comment:
what would happen if this mastersRole is not provided in the props?

Reply:
At current implementation, if `props.mastersRole` is not provided, aws-eks would NOT add additional iam principal as `system:masters` to the ConfigMap and only the cluster creator role would be `system:masters`. Now if you need to administer the cluster using a common IAM role with `kubectl`, you will need to assume to that cluster creator role and that could be a permission concern. The reason `prop.mastersRole` exists is that it allow users to pass any existing role to that property and aws-eks would add that into its ConfigMap with `system:masters` permission. If undefined, no additional ConfigMap entry would be added.

Now, with the introduction of access entry with access api, we essentially keep this support but we have to determine the AuthenticationMode in the following conditions:

1. AuthenticationMode is `API` - this means ConfigMap not supported anymore and we should not do this.cluster.awsAuth.addMasters() to that, instead, we should create AccessEntry for the passed mastersRole.
2. AuthenticationMode is `API_AND_CONFIG_MAP` - this means both are supported. This PR would leave everything as before and continue update the ConfigMap using awsAuth.addMastersRole().
3. 2. AuthenticationMode is `CONFIG_MAP`  - same as option 2.

Reply:
Understood, Thank you @pahud for explaining it well